clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleZonalServerGroupCachingAgent.groovy

```diff
@@ -134,14 +134,15 @@ class GoogleZonalServerGroupCachingAgent extends AbstractGoogleCachingAgent impl
   }
 
   private List<GoogleServerGroup> constructServerGroups(ProviderCache providerCache, String onDemandServerGroupName = null) {
+    GoogleZonalServerGroupCachingAgent cachingAgent = this
     List<String> zones = credentials.getZonesFromRegion(region)
     List<GoogleServerGroup> serverGroups = []
 
     BatchRequest igmRequest = buildBatchRequest()
     BatchRequest instanceGroupsRequest = buildBatchRequest()
     BatchRequest autoscalerRequest = buildBatchRequest()
 
-    List<InstanceTemplate> instanceTemplates = fetchInstanceTemplates(compute, project)
+    List<InstanceTemplate> instanceTemplates = fetchInstanceTemplates(this, compute, project)
```

Comment:
I thin kyou meant cachingAgent

Reply:
Good catch; technically it doesn't matter here as we're not in a closure, but since I've assigned the variable anyway (for the later closure) it's clearer to use it.